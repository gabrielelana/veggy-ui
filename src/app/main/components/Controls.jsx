const Controls = props => (
  <div className="column">
    <div className="box">
      <div className="control is-grouped ">
        <div className="control">
          <button id="startButton" disabled={props.startDisabled} onClick={props.onStart} className="button is-primary is-large">Start</button>
        </div>
      </div>
    </div>
  </div>
)


Controls.defaultProps = {
  startDisabled: false,
  onStart: () => {}
}

export default Controls