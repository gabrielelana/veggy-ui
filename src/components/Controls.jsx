import React from 'react'

const Controls = React.createClass({
  getDefaultProps() {
    return {
      startDisabled: false,
      squashDisabled: true,
      onStart: () => {},
      onSquash: () => {}
    }
  },
  render() {
    return (
      <div className="column">
        <div className="box">
          <div className="control is-grouped ">
            <div className="control">
              <button id="startButton" disabled={this.props.startDisabled} onClick={this.props.onStart} className="button is-primary is-large">Start</button>
            </div>
            <div className="control">
              <button id="squashButton" disabled={this.props.squashDisabled} onClick={this.props.onSquash} className="button is-danger is-large">Squash</button>
            </div>
          </div>
        </div>
      </div>
      )
  }
})

export default Controls