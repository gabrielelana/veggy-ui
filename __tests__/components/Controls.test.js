import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import Controls from '../../src/app/main/components/Controls'

describe('<Controls />', () => {
  
  test('with default props', () => {
    const component = renderer.create(<Controls />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('with startDisabled', () => {
    const component = renderer.create(<Controls startDisabled />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('with squashDisabled', () => {
    const component = renderer.create(<Controls squashDisabled />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('When start is clicked onStart callback should be called', (done) => {
    const component = shallow(<Controls onStart={() => { done() }} />)
    component.find('#startButton').simulate('click')
  })

  it('When squash is clicked onSquash callback should be called', (done) => {
    const component = shallow(<Controls squashDisabled={false} onSquash={() => { done() }} />)
    component.find('#squashButton').simulate('click')
  })
})