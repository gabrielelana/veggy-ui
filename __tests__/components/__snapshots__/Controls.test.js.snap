// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> with default props 1`] = `
<div
  className="column"
>
  <div
    className="box"
  >
    <div
      className="control is-grouped "
    >
      <div
        className="control"
      >
        <button
          className="button is-primary is-large"
          disabled={false}
          id="startButton"
          onClick={[Function]}
        >
          Start
        </button>
        <button
          className="button is-danger is-large"
          disabled={true}
          id="squashButton"
          onClick={[Function]}
        >
          Squash
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Controls /> with squashDisabled 1`] = `
<div
  className="column"
>
  <div
    className="box"
  >
    <div
      className="control is-grouped "
    >
      <div
        className="control"
      >
        <button
          className="button is-primary is-large"
          disabled={false}
          id="startButton"
          onClick={[Function]}
        >
          Start
        </button>
        <button
          className="button is-danger is-large"
          disabled={true}
          id="squashButton"
          onClick={[Function]}
        >
          Squash
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Controls /> with startDisabled 1`] = `
<div
  className="column"
>
  <div
    className="box"
  >
    <div
      className="control is-grouped "
    >
      <div
        className="control"
      >
        <button
          className="button is-primary is-large"
          disabled={true}
          id="startButton"
          onClick={[Function]}
        >
          Start
        </button>
        <button
          className="button is-danger is-large"
          disabled={true}
          id="squashButton"
          onClick={[Function]}
        >
          Squash
        </button>
      </div>
    </div>
  </div>
</div>
`;
